calculator/main.py

```diff
@@ -26,7 +28,23 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if operation not in get_operations(calc):
+        raise InvalidOperation(operation)
+    
+    check_params = all(isinstance(x,(int,float)) for x in params)
```

Comment:
🙌 Pythonic! Nice!

Please use spaces to separate arguments: https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements